app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt

```diff
@@ -100,13 +100,13 @@ class ParseSusiResponseHelper {
                     }
                     val map2 = tableanswer.data
                     Timber.d(map2.toString())
-                    val iterator2 = map2?.iterator()
-                    while (iterator2?.hasNext().toString().toBoolean()) {
-                        val entry2 = iterator2?.next()
+                    val iterator2 = map2.iterator()
+                    while (iterator2.hasNext().toString().toBoolean()) {
```

Comment:
iterator2.hasNext().toString().toBoolean().toString().toInt().toString().toBoolean()

Also, why not directly iterate via a forEach loop?

Reply:
while writing this I just went on with the while loop, is it more better using a for each loop ?

Reply:
Yes